src/common/base-store.ts

```diff
@@ -20,13 +19,18 @@ export interface BaseStoreParams<T = any> extends ConfOptions<T> {
  */
 export abstract class BaseStore<T = any> extends Singleton {
   protected storeConfig?: Config<T>;
-  protected syncDisposers: Function[] = [];
+  protected syncDisposers: Disposer[] = [];
 
-  whenLoaded = when(() => this.isLoaded);
   @observable isLoaded = false;
 
+  get whenLoaded() {
+    return when(() => this.isLoaded);
```

Comment:
Seems like `when` evaluates callback to detect observables before `makeObservable(this)` is done in `constructor`.

Reply:
Why do you think this? That getter shouldn't be called until after that `makeObservable(this)` executes.

Reply:
I was talking about how it was done before.